app/java-2/src/main/java/com/google/ProcessingJob.java

```diff
@@ -0,0 +1,142 @@
+        }
+    }
+
+    private static void throwErrorIfInvalidArgs() {
+        if (RAW_DATA_BUCKET == null || RAW_DATA_BUCKET.isEmpty()) {
+            throw new IllegalArgumentException("RAW_DATA_BUCKET required");
+        }
+        if (PROCESSED_DATA_BUCKET == null || PROCESSED_DATA_BUCKET.isEmpty()) {
+            throw new IllegalArgumentException("PROCESSED_DATA_BUCKET required");
+        }
+    }
+
+    public static String downloadRawData() throws IOException {
+        System.out.println("downloadRawData: start downloading data");
+
+        // Create temporary folder and file
+        Path tempDir = Files.createTempDirectory("rawData");
+        File tempDataFile = new File(tempDir.toFile(), "raw_data.csv");
+        System.out.println(
+            "downloadRawData: processing from " + RAW_DATA_BUCKET +
+                " to " + PROCESSED_DATA_BUCKET);
+
+        // Read from Google Cloud Storage bucket
+        Storage storage = StorageOptions.getDefaultInstance().getService();
+        Blob blob = storage.get(RAW_DATA_BUCKET, RAW_DATA_FILE);
+        blob.downloadTo(tempDataFile.toPath());
+        System.out.println("downloadRawData: downloaded data to " + tempDataFile.getAbsolutePath());
+        return tempDataFile.getAbsolutePath();
+    }
+
+    public static Dictionary<String, Integer> processRawData(String tempDataFilePath) 
+        throws FileNotFoundException, IOException {
+        System.out.println("processRawData: start processing data");
+        Dictionary<String, Integer> colorAgeLocationToCount = new Hashtable<String, Integer>();
+        int numOfProcessedRows = 0;
+        int numOfIgnoredRows = 0;
+        // Go through each row of the CSV file.
+        FileReader reader = new FileReader(tempDataFilePath);
+        CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT.withHeader());
+        for (CSVRecord record : csvParser) {
+            numOfProcessedRows++;
+            // Ignore rows where crucial columns are empty/unknown.
+            boolean shouldIgnoreRow = false;
+            for (String column : CSV_COLUMNS) {
+                String value = record.get(column);
+                if (value == null || value.isEmpty() || value.equals("?")) {
+                    numOfIgnoredRows++;        
+                    shouldIgnoreRow = true;
+                }
+            }
+
+            // Bump count in "Color/Age/Location" dictionary.
+            if (!shouldIgnoreRow) {
+                String key = String.format(
+                    "%s/%s/%s",
+                    record.get("Primary Fur Color"),
+                    record.get("Age"),
+                    record.get("Location")
+                );
+                Integer currCount = colorAgeLocationToCount.get(key);
+                if (currCount == null) {
+                    currCount = 0;
+                }
+                colorAgeLocationToCount.put(key, currCount + 1);
```

Comment:
*Issue*: I realize the structure of my JSON object is incorrect.
That is the JSON "objects" are just integers:
```
4
```
But they should actually look like:
```
{
    "Running": 2,
    "Chasing": 1,
    ...
    "_count": 4,
}
```
I'll fix that. :)

Reply:
Fixed.
See [`bumpCountsBasedOnCsvRow()`](https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/pull/32/commits/fad87474a2720cdccee7e436b2e19b83704224dc).